MoreLinq/Traverse.cs

```diff
@@ -0,0 +1,97 @@
+#region License and Terms
+// MoreLINQ - Extensions to LINQ to Objects
+// Copyright (c) 2016 Felipe Sateler. All rights reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#endregion
+namespace MoreLinq
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+
+    public partial class MoreEnumerable
+    {
+        /// <summary>
+        /// Generates a sequence by traversing an element according to the <paramref name="nextSteps"/> generator function. 
+        /// The returned elements are traversed Breadth First.
```

Comment:
> The returned elements are traversed Breadth First.

This reads confusing because it says how the user of the result will traverse the result whereas the returned elements are the results of a breadth-first traversal. How about this?

> The returned elements are the result of breadth-first traversal.